lib/internal/event_target.js

```diff
@@ -362,13 +369,21 @@ class Listener {
       previous.next = this;
     this.previous = previous;
     this.listener = listener;
-    // TODO(benjamingr) these 4 can be 'flags' to save 3 slots
-    this.once = once;
-    this.capture = capture;
-    this.passive = passive;
-    this.isNodeStyleListener = isNodeStyleListener;
+
+    let flags = 0b0;
+    if (once) 
+      flags |= FLAG_ONCE;
+    if (capture) 
+      flags |= FLAG_CAPTURE;
+    if (passive)
+      flags |= FLAG_PASSIVE;
+    if (isNodeStyleListener)
+      flags |= FLAG_NODE_STYLE;
+    if (Boolean(weak))
+      flags |= FLAG_WEAK;
+    this.flags = flags;
```

Comment:
```suggestion
    this.flags = 0 |
      (once && FLAG_ONCE) |
      (capture && FLAG_CAPTURE) |
      (isNodeStyleListener && FLAG_NODE_STYLE) |
      (Boolean(weak) && FLAG_WEAK);
```

Reply:
This is a bit too clever for my taste because of the reliance on how && uses the righthand side for the value.

Also, why is weak wrapped (before and after) in a conversion?